tests/drivers/i2c/i2c_slave_api/src/main.c

```diff
@@ -142,6 +142,8 @@ void test_eeprom_slave(void)
 	int reg_addr1 = DT_REG_ADDR(DT_INST(1, atmel_at24));
 	const char *label_eeprom0 = DT_LABEL(DT_INST(0, atmel_at24));
```

Comment:
While this solution works, an approach that would be more realistic with how this should be done is to add distinct node labels to the EEPROM devicetree nodes and use those.  That would eliminate any dependency on which device is assigned which instance.

However, when I do that on the branch immediately before this commit, the test still fails.  Which suggests that the test is wrong.  Which it is.  Multiple calls are using the wrong device address for the I2C controller.

Reply:
This comment is out of date; what it's trying to convey is better described down at the code that tries to figure out which device is which.